### PR TITLE
fix: dsa & dsi sending

### PR DESCRIPTION
**Pull request by Syn-McJ**
## Issue being fixed or feature implemented
Some bugs are preventing sending dsa & dsi

## What was done?
- Bug fixes
DashSync: https://github.com/dashpay/dashsync-iOS/pull/556

## How Has This Been Tested?
N/A

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: dash-spv-apple-bindings/src/coinjoin.rs [hunk 1]
```diff
@@ -117,16 +107,15 @@ pub unsafe extern "C" fn add_client_queue_manager(
     context: *const c_void
 ) {
     let client_queue_manager = CoinJoinClientQueueManager::new(
-        Rc::new(RefCell::new(std::ptr::read(client_manager_ptr))),
```
Comment (Syn-McJ) [inline dash-spv-apple-bindings/src/coinjoin.rs:120]: `std::ptr::read` creates a copy of client_manager, which is wrong. Using the raw pointer instead

File: dash-spv-coinjoin/src/coinjoin_client_session.rs [hunk 2]
```diff
@@ -775,15 +777,16 @@ impl CoinJoinClientSession {
         };
         
         let current_time = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs();
-        let is_timeout = self.base_session.time_last_successful_step + timeout + lag_time >= current_time;
+        let is_timeout = current_time >= self.base_session.time_last_successful_step + timeout + lag_time;
```
Comment (Syn-McJ) [inline dash-spv-coinjoin/src/coinjoin_client_session.rs:780]: Bad time check

File: dash-spv-coinjoin/src/utils/transaction_builder.rs [hunk 3]
```diff
@@ -82,8 +83,8 @@ impl<'a> TransactionBuilder {
                 input_hash: coin.tx_outpoint.hash,
                 index: coin.tx_outpoint.index,
                 script: None,
-                signature: None,
-                sequence: 0
+                signature: Some(Vec::new()),
```
Comment (Syn-McJ) [inline dash-spv-coinjoin/src/utils/transaction_builder.rs:86]: null signature was the actual problem with encoding DSI message

**Review by pankcuf — APPROVED** [review]
LGTM

**Pull request merged**